src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeeded();
+        elements[size] = value;
+        size++;
```

Comment:
```suggestion
        elements[size++] = value;
```
not a mistake, but a cool shortcut